test/mlir/onnx/onnx_shape_inference.mlir

```diff
@@ -164,6 +133,58 @@ func.func @test_transpose(%arg0 : tensor<5x5x1x32xf32>) -> tensor<*xf32> {
 
 // -----
 
+//===----------------------------------------------------------------------===//
+/// Test shape inference for DFT.
+//===----------------------------------------------------------------------===//
+
+func.func @test_dft(%arg0: tensor<1x8x10x12xf32> , %arg1 : tensor<i32>) -> tensor<*xf32> {
+  %cst = "onnx.NoValue"() {value} : () -> none
+  %0 = "onnx.DFT"(%arg0, %arg1, %cst) : (tensor<1x8x10x12xf32>, tensor<i32>, none) -> tensor<*xf32>
+  "onnx.Return"(%0) : (tensor<*xf32>) -> ()
+
+// mlir2FileCheck.py
+// CHECK-LABEL:  func.func @test_dft
+// CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<1x8x10x12xf32>, [[PARAM_1_:%.+]]: tensor<i32>) -> tensor<*xf32> {
+// CHECK:           [[VAR_0_:%.+]] = "onnx.NoValue"() {value} : () -> none
+// CHECK:           [[VAR_1_:%.+]] = "onnx.DFT"([[PARAM_0_]], [[PARAM_1_]], [[VAR_0_]]) {inverse = 0 : si64, onesided = 0 : si64} : (tensor<1x8x10x12xf32>, tensor<i32>, none) -> tensor<*xf32>
+// CHECK:           onnx.Return [[VAR_1_]] : tensor<*xf32>
```

Comment:
What did you test here? the output tensor is still unranked, e.g. `tensor<*xf32>`? The old code produced a correct output `tensor<1x8x10x12x2xf32>`, why is the output now unranked?

Same for the remaining lit tests.

Reply:
Because DFTV17 and DFTV20 are not a direct 1x1 mapping like we initially assumed. I am going to have to eventually separate them out like the code I had before and remove the decomposition. The conditions for DFTV17 is not the same for DFVT20 and so the results that are produced vary slightly (I spoke with @chentong319 about this). Also when I do attempt to have the output as `tensor<1x8x10x12x2xf32>`.... I get a failure when running `check-onnx-lit`. What you are seeing is the suggestions from the `/utils/fixLitTest.py ` script. That is the reasoning:

```
/Users/meganhampton/zDLC/onnx-mlir/test/mlir/onnx/onnx_shape_inference.mlir:163:16: error: CHECK-SAME: expected string not found in input
// CHECK-SAME: ([[PARAM_0_:%.+]]: tensor<1x8x10x12xf32>, [[PARAM_1_:%.+]]: tensor<i32>) -> tensor<1x8x10x12xf32> {
               ^
<stdin>:88:31: note: scanning from here
 func.func @test_dft_one_sided(%arg0: tensor<1x8x10x12xf32>, %arg1: tensor<i32>) -> tensor<*xf32> {
                              ^
/Users/meganhampton/zDLC/onnx-mlir/test/mlir/onnx/onnx_shape_inference.mlir:179:16: error: CHECK-SAME: expected string not found in input
// CHECK-SAME: ([[PARAM_0_:%.+]]: tensor<1x8x10x12xf32>, [[PARAM_1_:%.+]]: tensor<i32>) -> tensor<1x8x10x12xf32> {
               ^
<stdin>:97:29: note: scanning from here
 func.func @test_dft_inverse(%arg0: tensor<1x8x10x12xf32>, %arg1: tensor<i32>) -> tensor<*xf32> {

```

Reply:
If you have suggestions on how to improve the tests, please share. Thank you!

Reply:
Again, what did you want to test with these lit tests?

Reply:
kk I removed the lit tests. Thank you!